server/integration/event-logger/src/main/java/org/infinispan/server/eventlogger/ServerEventLogger.java

```diff
@@ -89,26 +99,35 @@ TimeService getTimeService() {
    }
 
    @Override
-   public List<EventLog> getEvents(int start, int count, Optional<EventLogCategory> category, Optional<EventLogLevel> level) {
-      QueryBuilder query = getQueryFactory().from(ServerEventImpl.class).orderBy("when", SortOrder.DESC).maxResults(count).startOffset(start);
-      if (category.isPresent()) {
-         if (level.isPresent()) {
-            query.having("category").eq(category.get()).and().having("level").eq(level.get());
-         } else {
-            query.having("category").eq(category.get());
+   public List<EventLog> getEvents(Instant start, int count, Optional<EventLogCategory> category, Optional<EventLogLevel> level) {
+      List<EventLog> events = new ArrayList<>();
+      ConcurrentMap<Address, Throwable> throwables = CollectionFactory.makeConcurrentMap();
+      try {
+         cacheManager.executor().submitConsumer(m -> {
+            Cache<Object, Object> cache = m.getCache(EVENT_LOG_CACHE);
+            QueryFactory<Query> queryFactory = Search.getQueryFactory(cache);
+            QueryBuilder query = queryFactory.from(ServerEventImpl.class).orderBy("when", SortOrder.DESC).maxResults(count);
+            FilterConditionContext filter = query.having("when").lte(start);
+            FilterConditionContext categoryFilter = category.map(c -> filter.and(queryFactory.having("category").eq(c))).orElse(filter);
```

Comment:
Thinking we don't need all these extra variables.  Pretty sure you can just reassign back to filter for this and next line.